core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -798,8 +799,8 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean) extends Abstra
   val logRetentionTimeMillis = getLogRetentionTimeMillis
   val minInSyncReplicas = getInt(KafkaConfig.MinInSyncReplicasProp)
   val logPreAllocateEnable: java.lang.Boolean = getBoolean(KafkaConfig.LogPreAllocateProp)
-  val messageFormatVersion = getString(KafkaConfig.MessageFormatVersionProp)
-  val messageTimestampType = getString(KafkaConfig.MessageTimestampTypeProp)
+  val messageFormatVersion = ApiVersion(getString(KafkaConfig.MessageFormatVersionProp))
```

Comment:
Currently, messageFormatVersion defaults to latest api version. For easier upgrade, perhaps it's better to default it to inter.broker.protocol version. Otherwise, users have to make sure these two values are consistent.